### PR TITLE
Updated dependencies

### PR DESCRIPTION
**Pull request by rachid-o**
Hi @herrvendil 

This branch is based upon the one of [this pull request](https://github.com/lightningj-org/lightningj/pull/76) so it contains both changes. Maybe it saves time to do the testing for both changes at once. If so, we could close the other one and only merge this one.

Again, tested it locally by in a separate project by retrieving a list of channels, invoices and payments.

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -16,31 +16,29 @@ ext {
 
 dependencies {
     //gRPC
-    compile 'io.netty:netty-tcnative-boringssl-static:2.0.54.Final'
-    compile "io.grpc:grpc-netty:$grpcVersion"
-    compile "io.grpc:grpc-protobuf:$grpcVersion"
-    compile "io.grpc:grpc-stub:$grpcVersion"
+    implementation 'io.netty:netty-tcnative-boringssl-static:2.0.54.Final'
```
Comment (herrvendil) [inline buildSrc/build.gradle:19]: Changing from compile -> implementation is necessary but the dependency management isn't handled exactly the same. I guess I need to update the documentation about required dependencies.

Do you have a list of dependencies you needed to define in your projects build.gradle in order to get all dependencies to resolve?

**Review by herrvendil — COMMENTED** [review]
I guess it is time to update the gradle wrapper as well.

**Comment by rachid-o** [comment]
> 



> I guess it is time to update the gradle wrapper as well.

Yes, I have that in a separate branch now. But since it broke some stuff, I thought let's first do the preparations before upgrading the wrapper itself.

File: buildSrc/build.gradle (hunk 1 above)
Comment (rachid-o) [reply buildSrc/build.gradle:19]: I did not have to change anything in my test project.

In my test project now most of the dependencies are (transitively) added to the `runtimeClasspath`, but not to `compileClasspath` anymore. Which is fine for my test project.

**Comment by herrvendil** [comment]
Hi again. 

I have continued to work on your PR with update to Gradle 7.5.1. I took some time since it was a lot of work to get everything working. You can find your changes pulled into https://github.com/lightningj-org/lightningj/pull/78

I have published a Snapshot to:
maven {url="https://oss.sonatype.org/content/repositories/snapshots/"}

And published a SNAPSHOT dependency:
implementation 'org.lightningj:lightningj:0.16.0-Beta-SNAPSHOT'

So you can try it out. One change I had to do was to change:
implementation "io.grpc:grpc-netty-shaded:$grpcVersion"
Into
api "io.grpc:grpc-netty-shaded:$grpcVersion"
in order for SSL Context to work (otherwise class not found error).

I was thinking to merge this into the upcomming 1.6 release of LND and leave 1.5.x as is. Since it only should be minor changes. What do you think?

**Comment by herrvendil** [comment]
Had to update the jaxb libraries as well in order to get it to work jdk 11+, so the valid snapshot is:
implementation 'org.lightningj:lightningj:0.16.0.1-Beta-SNAPSHOT'

**Comment by rachid-o** [comment]
I tested the version `0.16.0-Beta-SNAPSHOT` and it works fine in my test project too. But still this is a rather small project as I didn't caught the issue you encountered with the SSL Context. 

Sounds fine to release it with the next version. Shall we close my PR's then?
Just note that the current official version is broke for me, so I can only use the version I built myself or your snapshot version.

**Comment by herrvendil** [comment]
Ok maybe i should Merge and create a 1.5.3 release with the new updates anyway then so it works for you. Will do it in a couple of hours.

But you can close your PR since everything is in the new one.


Skickat från min iPhone

18 okt. 2022 kl. 21:33 skrev Rachid ***@***.***>:

﻿

I tested the version 0.16.0-Beta-SNAPSHOT and it works fine in my test project too. But still this is a rather small project as I didn't caught the issue you encountered with the SSL Context.

Sounds fine to release it with the next version. Shall we close my PR's then?
Just note that the current official version is broke for me, so I can only use the version I built myself or your snapshot version.

**Comment by rachid-o** [comment]
These changes are included in https://github.com/lightningj-org/lightningj/pull/78

**Pull request closed without merging**